ccp/lib/clients/openshift/client.py

```diff
@@ -0,0 +1,342 @@
+        :param namespace: The namespcae from which the BuildConfig is to be
+        removed
+        :param bc: The name of the BuildConfig to delete
+        :type namespace str
+        :type bc str
+        :return: output of executed command
+        :rtype str
+        :raises Exception
+        :raises subprocess.CalledProcessError
+        :raises ccp.lib.exceptions.CommandOutputError
+        """
+        command = "{base_command} delete -n {namespace} bc {bc}".format(
+            base_command=self.base_command,
+            namespace=namespace,
+            bc=bc,
+        )
+        return run_command(cmd=command, shell=False)
+
+    @retry(tries=10, delay=3, backoff=2)
+    def list_build_configs(self, namespace, filter_out=None, selectors=None):
+        """
+        Lists all the BuildConfigs in specified namespace except those
+        requested to be filtered out.
+        :param namespace: The namespace from which to get the build configs.
+        :param filter_out: Default None: A list of BuildConfig names to filter
+        out.
+        :param selectors Filter by selectors, provides list of labels
+        :type namespace str
+        :type filter_out list
+        :type selectors dict
+        :return: A list of BuildConfigs in namespace, after filtering
+        :rtype list
+        :raises Exception
+        :raises subprocess.CalledProcessError
+        :raises ccp.lib.exceptions.CommandOutputError
+        """
+        if not filter_out or not isinstance(filter_out, list):
+            filter_out = []
+        selector_params = ""
+        for k, v in selectors:
+            selector_params = selector_params + " --selector {k}={v}".format(
+                k=k,
+                v=v
+            )
+        command = "{base_command} get bc -o name -n {namespace}{s}".format(
+            base_command=self.base_command,
+            namespace=namespace,
+            s=selector_params
+        )
+        out = run_command(cmd=command)
+        if not out.strip():
+            return []
+        out = out.strip().split('\n')
+        return [each for each in out if each not in filter_out]
+
+    def list_builds(self, namespace, status_filter=None, filter_out=None):
+        """
+        Lists builds in a particular namespace, after filtering by status and
+        name.
+        :param namespace: The namespace from which to get the builds
+        :param status_filter: Default None : List of statuses to look for. If
+        provided, only builds with specified statuses are returned.
+        :param filter_out: Default None : List of builds to filter out.
+        :type namespace str
+        :type status_filter list
+        :type filter_out list
+        :return: List of builds that matched all conditions
+        :rtype list
+        :raises Exception
+        :raises subprocess.CalledProcessError
+        :raises ccp.lib.exceptions.CommandOutputError
+        """
+        filter_str = ""
+        close_str = "{{end}}"
```

Comment:
`filter_str` and `close_str` only applies `if status_filter` - so define these two variables only `if status_file`  is true.

Reply:
Nope. filter_str is the only one what applies in case of status_filter. close_str applies either way. It is just number of instances of it that change.

Reply:
I know it looks more efficient. But it is simply easier to keep it as is.